spark/dl/src/test/scala/com/intel/analytics/bigdl/nn/RecurrentSpec.scala

```diff
@@ -27,6 +27,106 @@ import scala.collection.mutable.ArrayBuffer
+    val hiddenSize = 4
+    val inputSize = 5
+    val outputSize = 5
+    val batchSize = 4
+    val time = 2
+    val seed = 100
+    RNG.setSeed(seed)
+
+    val model = Sequential[Double]()
+      .add(Recurrent[Double](BatchNormParams())
+        .add(LSTMPeephole[Double](inputSize, hiddenSize)))
+      .add(TimeDistributed[Double](Linear[Double](hiddenSize, outputSize)))
+
+    println(model)
+
+    val input = Tensor[Double](batchSize, time, inputSize)
+    val gradOutput = Tensor[Double](batchSize, time, outputSize)
+
+    val output = model.forward(input)
+    val gradInput = model.backward(input, gradOutput)
+
+    println("add normalization")
+  }
+
+  "A Recurrent with GRU cell " should " add batchNormalization correctly" in {
+    val hiddenSize = 4
+    val inputSize = 5
+    val outputSize = 5
+    val batchSize = 4
+    val time = 2
+    val seed = 100
+    RNG.setSeed(seed)
+
+    val model = Sequential[Double]()
+      .add(Recurrent[Double](BatchNormParams())
+        .add(GRU[Double](inputSize, hiddenSize)))
+      .add(TimeDistributed[Double](Linear[Double](hiddenSize, outputSize)))
+
+    println(model)
+
+    val input = Tensor[Double](batchSize, time, inputSize)
+    val gradOutput = Tensor[Double](batchSize, time, outputSize)
+
+    val output = model.forward(input)
+    val gradInput = model.backward(input, gradOutput)
+
+    println("add normalization")
+  }
+
+  "A Recurrent with LSTM cell " should " add batchNormalization correctly" in {
+    val hiddenSize = 4
+    val inputSize = 5
+    val outputSize = 5
+    val batchSize = 4
+    val time = 2
+    val seed = 100
+    RNG.setSeed(seed)
+
+    val model = Sequential[Double]()
+      .add(Recurrent[Double](BatchNormParams())
+        .add(LSTM[Double](inputSize, hiddenSize)))
+      .add(TimeDistributed[Double](Linear[Double](hiddenSize, outputSize)))
+
+    println(model)
+
+    val input = Tensor[Double](batchSize, time, inputSize)
+    val gradOutput = Tensor[Double](batchSize, time, outputSize)
+
+    val output = model.forward(input)
+    val gradInput = model.backward(input, gradOutput)
+
+    println("add normalization")
+  }
+
+  "A Recurrent with SimpleRNN cell " should " add batchNormalization correctly" in {
+    val hiddenSize = 4
+    val inputSize = 5
+    val outputSize = 5
+    val batchSize = 4
+    val time = 2
+    val seed = 100
+    RNG.setSeed(seed)
+
+    val model = Sequential[Double]()
+      .add(Recurrent[Double](BatchNormParams())
+        .add(RnnCell[Double](inputSize, hiddenSize, Tanh[Double]())))
+      .add(TimeDistributed[Double](Linear[Double](hiddenSize, outputSize)))
+
+    println(model)
+
+    val input = Tensor[Double](batchSize, time, inputSize)
+    val gradOutput = Tensor[Double](batchSize, time, outputSize)
+
+    val output = model.forward(input)
+    val gradInput = model.backward(input, gradOutput)
+
+    println("add normalization")
+  }
+
```

Comment:
Could you also add a case for convlstm, since it currently doesn't have pretopology and has a different code path.